regulation-worker/internal/delete/batch/filehandler/gzip.go

```diff
@@ -0,0 +1,105 @@
+package filehandler
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+
+	"github.com/rudderlabs/rudder-server/regulation-worker/internal/model"
+)
+
+type GZIPLocalFileHandler struct {
+	records []byte
+	casing  Case
+}
+
+func NewGZIPLocalFileHandler(casing Case) *GZIPLocalFileHandler {
+	return &GZIPLocalFileHandler{
+		casing: casing,
+	}
+}
+
+func (h *GZIPLocalFileHandler) Read(_ context.Context, path string) error {
+	f, err := os.OpenFile(path, os.O_RDWR, 0o644)
+	if err != nil {
+		return fmt.Errorf("error while opening compressed file, %w", err)
+	}
+	defer f.Close()
+
+	gzipReader, err := gzip.NewReader(f)
+	if err != nil {
+		return fmt.Errorf("error while reading compressed file: %w", err)
+	}
+
+	byt, err := io.ReadAll(gzipReader)
+	if err != nil {
+		return fmt.Errorf("unable to read contents of local file: %w", err)
+	}
+
+	h.records = byt
+	return nil
+}
+
+func (h *GZIPLocalFileHandler) Write(_ context.Context, path string) error {
+	f, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o644)
+	if err != nil {
+		return fmt.Errorf("error while opening file, %w", err)
+	}
+
+	defer f.Close()
+
+	gw := gzip.NewWriter(f)
+	defer gw.Close()
+
+	_, err = gw.Write(h.records)
+	if err != nil {
+		return fmt.Errorf("error while writing cleaned & compressed data:%w", err)
+	}
+
+	return nil
+}
+
+func (h *GZIPLocalFileHandler) RemoveIdentity(ctx context.Context, attributes []model.User) error {
```

Comment:
Given that we have changed the logic of `RemoveIdentity`  and since this is going to be one of the most time consuming part of the loop during deletion. I strongly feel we should benchmark the old & new deletion logic. 

I have a feeling that if `sed` knows about all the users that needs to be matched for, it will do some kind of optimization do avoid literally iterating over each `OR` condition of pattern file and look for a match in each line.

Reply:
Based on the man of sed, -f is just reading the commands from the file and appending it to the list of commands to sed. -e which we have replaced it with does the same thing as well. 

The main approach we have changed is letting sed completely handle the file from which the data needs to be removed and we are doing it in memory. This is something which needs to be benchmarked and checked against.